### PR TITLE
Update code to use newer Django storages object. 

### LINKED ISSUE #1187: Django's `get_storage_class` is deprecated in 4.2 (Django 5.1 compat.)
Got the following warning running tests in Django 4.2:

```
/usr/local/lib/python3.11/site-packages/django/core/files/storage/__init__.py:32 RemovedInDjango51Warning: django.core.files.storage.get_storage_class is deprecated in favor of using django.core.files.storage.storages.
```

### PR DESCRIPTION
**Pull request by carltongibson**
Fixes #1187
Compare #1216 and #1202. 

Adds a new helper to call into Django's `storages` handler to look for the appropriate aliases, and populate those keys if not manually set. This allows users to opt-in to Django's new setting, but not require that they do anything for the default case. 

@diox can I ask you to take a glance, and let me know what you think. 

I need to do docs and change notes. **Update**: Done ✅

Q: From the release of 5.0, Django's official advice is to drop Django<4.2. Doing so would allow us to proceed more or less with how this is now. Keeping compat with older versions would require branching on availability as you see in #1202 particularly. I'm not sure that's worth it now: Anyone on an older Django can just use an older compressor, but I'm happy to follow your guidance there. (CI is passing for Django 4.2+)

### CODE REVIEW

**Comment by diox** [comment]
> Q: From the release of 5.0, Django's official advice is to drop Django<4.2. Doing so would allow us to proceed more or less with how this is now. Keeping compat with older versions would require branching on availability as you see in https://github.com/django-compressor/django-compressor/pull/1202 particularly. I'm not sure that's worth it now: Anyone on an older Django can just use an older compressor, but I'm happy to follow your guidance there. (CI is passing for Django 4.2+)

I'm fine with dropping Django < 4.2 support. compressor is fairly stable, I agree users on older Django can keep using the old version.

File: compressor/storage.py [hunk 1]
```diff
@@ -4,12 +4,25 @@
...
+def get_storage(
+    alias=settings.COMPRESS_STORAGE_ALIAS,
+    storage_class=settings.COMPRESS_STORAGE
+):
+    try:
+        return storages[alias]
+    except InvalidStorageError:
+        storages.backends[alias] = {
+            "BACKEND": storage_class
+        }
```
Comment (diox) [inline compressor/storage.py:22]: Modifying `django.core.files.storage.storages` ? That feels kinda wrong....

Comment (carltongibson) [reply compressor/storage.py:22]: I plan to request that this be made public API, just so that we have the stability guarantee.

Comment (carltongibson) [reply compressor/storage.py:22]: Forum thread for that: https://forum.djangoproject.com/t/document-storagehandler-as-public-api/32101

**Comment by carltongibson** [comment]
Hi @diox — I rebased after #1250 and added docs for the new settings and changelog entry. Can I ask you to take another look?

File: compressor/storage.py (hunk 1 above)
Comment (diox) [reply compressor/storage.py:22]: Even if `storages` dict ends up being documented it feels weird to me to modify it in an app, but I don't have a better API suggestion and I trust your judgment on that.

File: compressor/tests/test_storages.py [hunk 2]
```diff
@@ -15,16 +15,16 @@
 
 class GzipStorage(LazyObject):
     def _setup(self):
-        self._wrapped = get_storage_class(
-            "compressor.storage.GzipCompressorFileStorage"
-        )()
+        self._wrapped = storages.create_storage({
```
Comment (diox) [inline compressor/tests/test_storages.py:18]: That's not a documented API is it ? Same below.

Comment (carltongibson) [reply compressor/tests/test_storages.py:18]: No, not yet. Same as setting the backends on the StorageHandler, I want to make this documented. (We need it; so will others.)

File: compressor/storage.py (hunk 1 above)
Comment (carltongibson) [reply compressor/storage.py:22]: I'm open to another way of tying into the new API but other than forcing users to declare the storages we need in settings, which will almost never be customised, I don't see it. 🤔

"The StorageHandler is instantiated from settings, and then apps can add defaults to it if needed", is the story I'm going for.

Comment (carltongibson) [reply compressor/storage.py:22]: Opened a PR for that: https://github.com/django/django/pull/18277

**Review by diox — APPROVED** [review]
WFM if django internal bits get documented eventually...

**Comment by carltongibson** [comment]
Thanks @diox. 

FWIW, I was thinking about how we might back out of this if the current API is ruled private. (So not really safe, even though unlikely to change any time soon)

The advantage of the approach here is leaning into the new storages API. 

We could still do the initial `try` block. Then rather than going back to `storages` we could create the storage and return it directly. We'd loose the instance caching but 🤷‍♀️

We'd need to vendor the create storage method code, but that would be doable. 

So I don't think we're in a dead end situation even if the discussion on Django goes the other way. 

I'd quite like to push this forward now though, so we can prepare a release before Django 5.1b1 that this week coming up. 

Sound reasonable?

**Comment by diox** [comment]
Yep sounds like a plan!

**Comment by carltongibson** [comment]
@diox I realise I need to ask you to make the release. As far as I can see it's bump the version number and set the date, as per 20306853e9e8e20163c7a002df49a6c4ed56e748. (And tag and push and that 🎁 Thanks.)

**Comment by diox** [comment]
Released https://pypi.org/manage/project/django-compressor/release/4.5/ (tag: https://github.com/django-compressor/django-compressor/tree/4.5)

**Comment by carltongibson** [comment]
Just to circle back, the docs change here was just merged into Django, so the API is now covered by the Stability Policy. 

https://github.com/django/django/pull/18277

**Pull request merged**